src/openforms/js/components/admin/form_design/registrations/objectsapi/LegacyConfigFields.js

```diff
@@ -183,7 +183,10 @@ const LegacyConfigFields = ({index, name, schema, formData, onFieldChange}) => {
           id="root_uploadSubmissionCsv"
           name="uploadSubmissionCsv"
           value={uploadSubmissionCsv}
-          onChange={onFieldChange}
+          checked={uploadSubmissionCsv}
+          onChange={e =>
+            onFieldChange({target: {name: 'uploadSubmissionCsv', value: !uploadSubmissionCsv}})
```

Comment:
I chose to go this way instead of updating `onFieldChange ` https://github.com/open-formulieren/open-forms/blob/master/src/openforms/js/components/admin/form_design/registrations/objectsapi/ObjectsApiOptionsFormFields.js#L70 with nested if statements for possible boolean values. If you think it would be better I can change this.

Reply:
This looks better to me as well